### PR TITLE
www-servers/tomcat: fix startup via catalina.sh

### PR DESCRIPTION
**Pull request by hasufell**
@gentoo/java 

revbump, because installed files change

This fixes starting tomcat for me via
```sh
/usr/share/tomcat-8/bin/catalina.sh run
```

before that, it fails with
```
26-Oct-2015 19:31:32.826 WARNING [main] org.apache.catalina.startup.ClassLoaderFactory.validateFile Problem with directory [/${gentoo.classpath}], exists: [false], isDirectory: [false], canRead: [false]
26-Oct-2015 19:31:33.052 SEVERE [main] org.apache.tomcat.util.digester.Digester.startElement Begin event threw error
 java.lang.NoClassDefFoundError: javax/servlet/http/HttpServletRequest

[... more java errors ...]
```

ebuild diff:
```diff
--- tomcat-8.0.28-r1.ebuild	2015-10-26 20:16:10.218085628 +0100
+++ tomcat-8.0.28-r2.ebuild	2015-10-26 20:27:34.014052452 +0100
@@ -55,7 +55,7 @@
 
 	# For use of catalina.sh in netbeans
 	sed -i -e "/^# ----- Execute The Requested Command/ a\
-		CLASSPATH=\`java-config --classpath ${PN}-${SLOT}\`" \
+		CLASSPATH=\`java-config --with-dependencies --classpath ${PN}-${SLOT}\`" \
 		bin/catalina.sh || die
 }
 
```
Also applied to stable ebuild (and revbumped).

### CODE REVIEW

File: www-servers/tomcat/tomcat-8.0.28-r2.ebuild [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+	### Config ###
+
+	# create "logs" directory in $CATALINA_BASE
+	# and set correct perms, see #458890
+	dodir "${dest}"/logs
+	fperms 0750 "${dest}"/logs
+
+	# replace the default pw with a random one, see #92281
+	local randpw=$(echo ${RANDOM}|md5sum|cut -c 1-15)
```
Comment (mgorny) [inline www-servers/tomcat/tomcat-8.0.28-r2.ebuild]: NITPICK: This one has really bad entropy, man. Figure out something better :-P. Plus relying on GNU tools is not very portable.

**Comment by hasufell** [comment]
I think this also affects stable arch.

**Comment by mgorny** [comment]
@hasufell, found you some irrelevant thing to fix!

**Comment by hasufell** [comment]
but I am not the maintainer!!! I will set up a sieve filter and block you.

**Comment by hasufell** [comment]
@chewi @monsieurp :dancers: :mailbox:

**Comment by chewi** [comment]
I'm afraid this will end up having the same problem as [bug #453212](https://bugs.gentoo.org/show_bug.cgi?id=453212). I didn't notice this sed call when fixing that bug and I should have given it the same treatment. It should look something like how we now set the CLASSPATH in tomcat-r1.init. I need to clear my own backlog before I can look at this. Feel free to have a crack at it yourself.

**Comment by hasufell** [comment]
@chewi you can at least cherry-pick https://github.com/hasufell/gentoo/commit/fcdf96eee73086c526c6aafdff49c09c6630b1ef

**Comment by chewi** [comment]
Done.

**Comment by mgorny** [comment]
So what do we do about this PR?

**Pull request closed without merging**